js/src/dataGrid/cell/renderer/BeakerXCellRenderer.ts

```diff
@@ -159,18 +158,23 @@ export abstract class BeakerXCellRenderer extends TextRenderer {
   }
 
   getTextColor(config): string {
-    if (config.region === 'row-header') {
-      return Theme.DEFAULT_DATA_FONT_COLOR;
+    if (
+      config.region === 'row-header' ||
+      config.region === 'column-header' ||
+      config.region === 'corner-header'
+    ) {
+      return Theme.DEFAULT_HEADER_FONT_COLOR;
     }
-
-    const dataFontColor =
-      this.dataGrid.rowManager.rows[config.row] && this.dataGrid.rowManager.rows[config.row].cells
-        ? DataGridStyle.formatColor(this.dataGrid.rowManager.rows[config.row].cells[config.column].fontColor)
-        : Theme.DEFAULT_DATA_FONT_COLOR;
-
-    return config.region === 'column-header' || config.region === 'corner-header'
-      ? Theme.DEFAULT_HEADER_FONT_COLOR
-      : dataFontColor;
+    
+    if (
+      config.region === 'body' &&
+      this.dataGrid.rowManager.rows[config.row] &&
+      this.dataGrid.rowManager.rows[config.row].cells
+    ) {
+      return DataGridStyle.formatColor(this.dataGrid.rowManager.rows[config.row].cells[config.column].fontColor);
```

Comment:
This `formatColor` will be executed every single time we draw a table cell. Do you think we could do the format only once?

Reply:
Maybe we could save the color formatted.

Reply:
Let's discard this comment, it was already like this